libcaf_core/caf/flow/observer.hpp

```diff
@@ -338,13 +338,14 @@ class buffer_writer_impl : public detail::atomic_ref_counted,
   }
 
   void on_next(const value_type& item) override {
-    CAF_LOG_TRACE(CAF_ARG(item));
+    auto exit_guard = log::core::trace("item = {}",
+                                       const_cast<const value_type*>(&item));
```

Comment:
const_cast was introduced in this log statement to fix the error where the build fails with `error C2440: 'static_cast': cannot convert from 'const T' to 'const void *'` for some data types like `caf::chunk`. 
Please, let me know if you have suggestion for a better alternative here.